ui/raidboss/emulator/overrides/RaidEmulatorTimelineController.js

```diff
@@ -43,7 +43,10 @@ export default class RaidEmulatorTimelineController extends TimelineController {
     e.detail.logs.forEach((line) => {
       this.activeTimeline.emulatedTimeOffset = line.offset;
       this.ui.emulatedTimeOffset = line.offset;
+      this.activeTimeline.timebase =
+        new Date() - (line.offset - this.emulator.currentEncounter.encounter.initialOffset);
```

Comment:
Maybe this is a continuation of the suppressSeconds conversation from the other patch, but it is weird to me to ever see `new Date()` in any emulator code.  My expectation is that time should be some "virtual time" entirely controlled by the emulator.

It looks like `new Date()` is used in `SyncTo` and `_OnUpdateTimer()`.

What if:
* `activetimeline.OnLogLine` also took a current time parameter (the `SyncTo` case)
* `_OnUpdateTimer` also took a current time parameter
* update `timeline.js` to pass `new Date()` to the callers of these two functions
* Pass the log line time to `OnLogLine`, which will set the timebase for you
* Call `_OnUpdateTimer` with the current log line time as well

Reply:
Any time there's a `new Date` call that's not isolated, I have to account for it with extra logic like this in emulator. That's why I was suggesting that calls to `new Date` in raidboss be replaced with calls to a class function `getDate` or something that could be overridden in emulator as a global "return the current log line's timestamp" logic.

Reply:
I agree that `new Date()` needs to be abstracted.  In general, I prefer passing in the time explicitly to functions vs the implicit overriding of a functions.  It means that internal functions can't call `getDate` repeatedly (which is usually an anti-pattern for a processing function) and also makes it more clear that the date is being controlled explicitly externally and may not behave monotically.

Reply:
I think this would belong in a separate PR, so the question becomes should this PR wait until that change, or should this be merged as-is and the date stuff be a follow up PR?

Reply:
Oops, somehow I lost track of responding to this comment.  Thanks for submitting the other PR, you read my mind.  :)